server/src/main/java/org/opensearch/cluster/service/applicationtemplates/TemplateLoader.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service.applicationtemplates;
+
+import java.io.IOException;
+
+
+/**
+ * Interface to load template into the OpenSearch runtime.
+ */
+public interface TemplateLoader {
```

Comment:
SystemTemplateLoader, it may not make sense to use for other templates in general